ts/Extensions/OverlappingDataLabels.ts

```diff
@@ -278,9 +278,11 @@ function hideOrShow(label: SVGElement, chart: Chart): boolean {
                 isLabelAffected = true;
 
                 // Animate or set the opacity
-                label.alignAttr.opacity = newOpacity;
+                const labelAttr = label.dataLabelPosition?.posAttribs ||
+                    label.alignAttr;
+                labelAttr.opacity = newOpacity;
                 label[label.isOld ? 'animate' : 'attr'](
-                    label.alignAttr,
+                    labelAttr,
```

Comment:
Is a change in the core needed? Do we know why `label.alignAttr` is not updated for the funnel? Is it something specific to the funnel series, or can the same situation appear in a pie series?

Reply:
Thanks for the backticks suggestions, I changed the PR description. 

Funnel utilizes Pie drawDataLabels only when inside: false. 
https://github.com/highcharts/highcharts/blob/12a5a01b32a09882151b806e86021fe269b71019/ts/Series/Funnel/FunnelSeries.ts#L248-L254
Pie dataLabels are positioned based on `dataLabelPosition.posAttribs` and the `alignAttr` for pie label remains undefined (which means that it's position is never updated in `OverlappingDataLabels`) because alignAttr is only updated by `SVGElement.align` in `alignDataLabel` which is not utilized by pie labels but it is by funnel with `dataLabels.inside` = false and that's why the incorrect position is applied in `OverlappingDataLabels`.
https://github.com/highcharts/highcharts/blob/12a5a01b32a09882151b806e86021fe269b71019/ts/Series/Funnel/FunnelSeries.ts#L216-L226

Reply:
Thanks for the backticks suggestions @TorsteinHonsi , I changed the PR description. 

Funnel utilizes Pie drawDataLabels only when inside: false. 
https://github.com/highcharts/highcharts/blob/12a5a01b32a09882151b806e86021fe269b71019/ts/Series/Funnel/FunnelSeries.ts#L248-L254
Pie dataLabels are positioned based on `dataLabelPosition.posAttribs` and the `alignAttr` for pie label remains undefined (which means that it's position is never updated in `OverlappingDataLabels`) because alignAttr is only updated by `SVGElement.align` in `alignDataLabel` which is not utilized by pie labels but it is by funnel with `dataLabels.inside` = false and that's why the incorrect position is applied in `OverlappingDataLabels`.
https://github.com/highcharts/highcharts/blob/12a5a01b32a09882151b806e86021fe269b71019/ts/Series/Funnel/FunnelSeries.ts#L216-L226

Reply:
So can we override this somehow from the funnel series module so we don't need to make an exception in the overlapping labels module?

Reply:
Sure! Although, I haven't found a right place to overwrite the `alignAttr` in the funnel module after `posAttribs` is calculated and before utilizing `hideOrShow` where the position is updated so I had to update it in PieDataLabels.

Reply:
I found some inconsistencies and moved the fix to the funnel series (and simplified overlappping labels along the way). Thanks!